### PR TITLE
Integration/routing

### PR DESCRIPTION
**Pull request by msmcconnell**
# PR Details
## Description
This PR creates an integration/routing branch on carma_utils to support 
https://github.com/usdot-fhwa-stol/carma-platform/pull/947

A collection of new methods are added to the trajectory_utils package to better support plugin development. 

## How Has This Been Tested?
See referenced PR

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: trajectory_utils/src/trajectory_utils/conversions/conversions.cpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  double prev_time = times[0];
+  speeds->push_back(prev_speed);
+  for (int i = 1; i < downtrack.size(); i++)
+  {
+    double cur_pos = downtrack[i];
+    double cur_time = times[i];
+    double dt = cur_time - prev_time;
+    double delta_d = cur_pos - prev_position;
+
+    double cur_speed = (2.0 * delta_d / dt) - prev_speed;
```
Comment (MishkaMN) [inline trajectory_utils/src/trajectory_utils/conversions/conversions.cpp:123]: Is this supposed to be 2?

**Review by MishkaMN — CHANGES_REQUESTED** [review]

File: trajectory_utils/src/trajectory_utils/conversions/conversions.cpp (hunk 1 above)
Comment (msmcconnell) [reply trajectory_utils/src/trajectory_utils/conversions/conversions.cpp:123]: From the kinematic equations you get 
d_x = ((v_i + v_f) / 2.0) * t
Rearranging to solve for v_f gives you

(2.0 * d_x / t) - v_i

Comment (MishkaMN) [reply trajectory_utils/src/trajectory_utils/conversions/conversions.cpp:123]: ah makes sense

**Review by MishkaMN — DISMISSED** [review]

File: carma_utils/include/carma_utils/containers/containers.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+namespace containers
+{
+/**
+ * \brief Downsamples an input vector by saving only each nth element.
+ * For example, given an input vector of { 0, 1, 2, 3, 4, 5 } and n = 2
+ * The output vector will be {0, 2, 4}
+ * 
+ * \param input The input vector to downsample
+ * \param n The count of the elements to save
+ * // TODO write unit tests for this
```
Comment (saina-ramyar) [inline carma_utils/include/carma_utils/containers/containers.h:34]: Is TODO still needed?

File: trajectory_utils/src/trajectory_utils/trajectory_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    double new_speed = cur_speed;
+    if (cur_speed > prev_speed)
+    {  // Acceleration case
+      new_speed = std::min(cur_speed, prev_speed + accel_limit * delta_t);
+    }
+    else if (cur_speed < prev_speed)
+    {  // Deceleration case
+      new_speed = std::max(cur_speed, prev_speed - decel_limit * delta_t);
+    }
+    else
```
Comment (saina-ramyar) [inline trajectory_utils/src/trajectory_utils/trajectory_utils.cpp:165]: Does not matter much but I suggest remove  "else" to be consistent with the previous function.

**Review by saina-ramyar — APPROVED** [review]

**Review by saina-ramyar — APPROVED** [review]

**Pull request merged**